packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -377,10 +383,20 @@ class ScaffoldMessengerState extends State<ScaffoldMessenger> with TickerProvide
 
   void _updateScaffolds() {
     for (final ScaffoldState scaffold in _scaffolds) {
-      scaffold._updateSnackBar();
+      if (_isRoot(scaffold))
+        scaffold._updateSnackBar();
```

Comment:
There can only be one root, right? So you could exit the for loop when you've found it.

Reply:
See comment above.